### DISCUSSION TITLE
Suddenly Could not resolve hostname github.com

### DISCUSSION THREAD

**Discussion by ChristianVermeulen** [community/community#31487] [Actions] [answered]
## ⚠️ BLOCKING
This is an extremely blocking issue for us since we can not deploy to our application at the moment! I have reached out to Github support, but no reply yet.

---

We have been running our deploys from github actions for almost 2 years now. Yesterday, out of the blue we started getting problems.

We deploy using Capistrano, which connects to the servers from an ubuntu github action runner to our servers and performs all tasks on the servers. The first step of capistrano is to validate the connection with the git repository to check for the latest commit has with:

```
git ls-remote git@github.com:itexpertsbv/aident.git HEAD
```

This has always worked perfectly, until yesterday. Since yesterday Capistrano fails with the following message:

```
ssh: Could not resolve hostname github.com: Temporary failure in name resolution
fatal: Could not read from remote repository.
```

I then proceeded to run the `git ls-remote`-command outside of capistrano to see if the problem was with capistrano itself. But is wasn't. When I add the command as first thing in my action workflow, the following appears:

```
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```

So now, instead of having a connection problem we seem to have a permission problem... However, the `actions/checkout@v1`-step runs just fine, so a connection ánd authentication is still possible.

I then proceeded to try and `ping github.com`. Which pretty much freezes my action. It gets stuck at that step and doesn't continue anymore, eating away my minutes.

Now, I'm not aware of configuring any key or something in my action in order to communicate over git. On the contrary, there should be a key injected into the action. Something must have changed on the Github side yesterday but I can't figure out what...

**Comment by airtower-luna** [comment, marked as answer]
> We deploy using Capistrano, which connects to the servers from an ubuntu github action runner to our servers and performs all tasks on the servers.

Do those happen to be (deprecated) Ubuntu 18.04 runners? I've read Azure had some issues with Ubuntu 18.04 VMs [due to a systemd bug](https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1988119) yesterday.

**Reply by ChristianVermeulen** [reply]
@airtower-luna  No, I use `runs-on: ubuntu-latest`, and according to the runner log, it is using 20.04.4.

```
2022-09-01T06:45:51.4180449Z ##[group]Operating System
2022-09-01T06:45:51.4181484Z Ubuntu
2022-09-01T06:45:51.4182110Z 20.04.4
2022-09-01T06:45:51.4182791Z LTS
2022-09-01T06:45:51.4183483Z ##[endgroup]
```

**Reply by ChristianVermeulen** [reply]
@airtower-luna I must stand corrected. Our azure instances actually are 18.04, but I logged in to those over SSH and ran the `git ls-remote` without any problems. So I determined there was no connectivity problem between our azure instances and the git repository.

It's beyond me why I am able to run the command manually (over ssh), but capistrano is not...

In any case, thanks for the help!

**Comment by whimsy-cat** [comment]
I restarted the Ubuntu system and it is working well.
